extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcServerProcessor.java

```diff
@@ -219,13 +222,16 @@ void discoverBindableServices(BuildProducer<BindableServiceBuildItem> bindables,
 
     private Set<MethodInfo> gatherBlockingMethods(ClassInfo service) {
         Set<MethodInfo> result = new HashSet<>();
-        boolean classHasBlocking = service.classAnnotation(GrpcDotNames.BLOCKING) != null;
+        // if we have @Transactional, and no @NonBlocking,
+        boolean blockingClass = service.classAnnotation(BLOCKING) != null
+                || (service.classAnnotation(TRANSACTIONAL) != null &&
+                        service.classAnnotation(NON_BLOCKING) == null);
```

Comment:
Something we should report, and not only in gRPC is when @Blocking and @NonBlocking are used at the same time on the same element. It's clearly illegal, and should be reported properly.

CC @geoand

Reply:
I think we do that already in RR, but I might be misremembering.
I'll check tomorrow and add it if we don't

Reply:
Fixed in https://github.com/quarkusio/quarkus/pull/19601

Reply:
I think it should be handled somewhere globally but Georgios does not so I added checks here ;)